src/serac/physics/state/state_manager.cpp

```diff
@@ -192,8 +192,6 @@ void StateManager::storeDual(FiniteElementDual& dual)
     // Create a new grid function with unallocated data. This will be managed by sidre.
     grid_function = new mfem::ParGridFunction(&dual.space(), static_cast<double*>(nullptr));
     datacoll.RegisterField(name, grid_function);
```

Comment:
We discussed this via chat while you were working on this, so this is just for reference: the grid function should be set from the dual values. Just mark this as resolved once you get that change pushed in.